parsl/providers/aws/aws.py

```diff
@@ -139,7 +139,7 @@ def __init__(self,
         self.launcher = launcher
         self.linger = linger
         self.resources = {}
-        self.state_file = state_file if state_file is not None else '.ec2_{}.json'.format(self.label)
+        self.state_file = state_file if state_file is not None else 'awsproviderstate.json'
```

Comment:
If a user specifies some other parameter for state_file, such as`AWSProvider(state_file="foo.json")`, then I think a similar bug still exists: `write_state_file` will still write to awsproviderstate.json, rather than to `foo.json`.

If that is the case, then maybe `write_state_file` should always write to self.state_file, rather than a hardcoded path?

Reply:
Thanks Ben, You're right. I just added another commit to write the state file to self.state_file and not the hardcoded path.